src/java.base/share/classes/sun/net/util/IPAddressUtil.java

```diff
@@ -479,10 +489,14 @@ private static String describeChar(char c) {
         return "'" + c + "'";
     }
 
-    private static String checkUserInfo(String str) {
+    // Check user-info component.
+    // This method returns an error message if a problem
+    // is found. The caller is expected to use that message to
+    // throw an exception.
+    public static String checkUserInfo(String str) {
```

Comment:
I think IPAddressUtil was original introduced as a utility class to support parsing of IP addresses and do text <--> byte[] conversion. It has grown into checking the authority and userInfo components so I think is begging for a rename or some refactoring. It's not this PR but I hope you see what I mean on the naming/purpose of this class.

Reply:
Yes - I see. I have logged https://bugs.openjdk.org/browse/JDK-8294951